### DISCUSSION TITLE
Absolute path for imports

### DISCUSSION THREAD

**Discussion by ghost** [preactjs/preact-cli#1541] [Q&A]
```typescript
// relative path
import Hello from '../../../components/hello'

// absolute path
import Hello from 'components/hello'
```

### Do you want to request a _feature_ or report a _bug_?
feature

Build failed!
_✖ ERROR ./index.ts_
> Module not found: Error: Can't resolve 'app' in './src'
 @ ./index.ts 1:0-30 1:0-30
 @ ../node_modules/preact-cli/lib/lib/entry.js
 @ multi ../node_modules/preact-cli/lib/lib/entry webpack-dev-server/client webpack/hot/dev-server

_✖ ERROR BabelEsmPlugin: ./index.ts_
> Module not found: Error: Can't resolve 'app' in './src'
 @ ./index.ts 1:0-30 1:0-30
 @ ../node_modules/preact-cli/lib/lib/entry.js
 @ multi ../node_modules/webpack-dev-server/client?http://0.0.0.0:8080 ../node_modules/webpack/hot/dev-server.js ../node_modules/preact-cli/lib/lib/ent
ry webpack-dev-server/client webpack/hot/dev-server**What is the current behaviour?**

### what is motivation or use case for changing the behaviour?
create-react-app has this functionality

### **Please mention other relevant information.**
I tried setting _tsconfig.json/_**baseUrl** to ```"./src/"```
I also tried [this](https://moduscreate.com/blog/es6-es2015-import-no-relative-path-webpack/):
```javascript
// <projectRoot>/webpack.config.js
{
  resolve: {
    modules: [
      path.resolve('./src'),
      path.resolve('./node_modules')
    ]
  }
}
```

**Comment by prateekbh** [comment]
Did you tried adding this via `preact.config.js`?

**Comment by ghost** [comment]
Thanks @prateekbh,
```javascript
config.resolve.modules.push(env.src)
```

And setting baseUrl in tsconfig.json fixed the vscode's _module not found_ issue

Are there any plans to make support for this by default?

**Comment by gitbreaker222** [comment]
Still getting the error in fresh `preact create default`:

> _./src/components/app.js_
Module not found: Error: Can't resolve 'src/routes/home' in '/<pathToProject>/src/components'

```javascript
// <projectRoot>/preact.config.js
export default (config, env, helpers) => {
  config.resolve.modules.push(env.src);
};
```

```javascript
// <projectRoot>/src/components/app.js

// ...

// Code-splitting is automated for routes
import Home from 'src/routes/home'; // '../../src/routes/home'

// ...
```

What am I missing?

-------

*edit:* found it! ` config.resolve.modules.push(env.src);` already starts in `src`

```javascript
import Home from 'src/routes/home'; // wrong
import Home from 'routes/home'; // right
```

**Comment by cubbie** [comment]
Sorry if this isn't the right spot but I feel like this is also affecting how my imports are working with TypeScript. After adding `config.resolve.modules.push(env.src);`
```
import Service from 'services/service'; // work but error ts(2307)
import Service from '@services/service'; // doesn't work but no lint error
```

**Comment by maltoze** [comment]
```javascript
// preact.config.js
module.exports = function (config, env) {
  config.resolve.alias.src = env.src;
};
```
this worked for me.

**Reply by nv08** [reply]
Worked for me
